pkg/meta/lib/meta.dart

```diff
@@ -130,6 +130,22 @@ const _Literal literal = const _Literal();
 ///   without invoking the overridden method.
 const _MustCallSuper mustCallSuper = const _MustCallSuper();
 
+/// Used to annotate a class that may not be no consumed as an interface.
+///
+/// This is useful because changes that would result in minor version
+/// increments require major version increments when applied to interfaces
+/// so packages may want to prevent consumers from implementing certain
+/// classes in order to protect semantic version guarantees.
+///
+/// Tools, such as the analyzer, can provide feedback if
+///
+/// * the annotation is associated with anything other than a class declaration
+/// * a class that has this annotation is used, either directly or
+///   transitively, as an interface (using `implements`)
+///
+/// At this time the analyzer does not enforce this annotation.
```

Comment:
Should this symbol be added in coordination with native dart analyzer support?

Reply:
There is a history of adding annotations before they are enforced by the analyzer, so the answer is "maybe", I think.

Reply:
It's fine to add an annotation before there is analyzer support, but we do not include the current status of the analyzer's implementation in the documentation for the annotation. This line should be removed.